airflow/contrib/operators/s3_copy_object_operator.py

```diff
@@ -0,0 +1,129 @@
+    :type verify: bool or str
+    """
+
+    @apply_defaults
+    def __init__(
+            self,
+            dest_bucket_key,
+            source_bucket_key,
+            dest_bucket_name=None,
+            source_bucket_name=None,
+            source_version_id=None,
+            s3_conn_id='aws_default',
+            verify=None,
+            *args, **kwargs):
+        super(S3CopyObjectOperator, self).__init__(*args, **kwargs)
+
+        if dest_bucket_name is None:
+            parsed_url = urlparse(dest_bucket_key)
+            if parsed_url.netloc == '':
+                raise AirflowException('Please provide dest_bucket_name')
+            else:
+                dest_bucket_name = parsed_url.netloc
+                if parsed_url.path[0] == '/':
+                    dest_bucket_key = parsed_url.path[1:]
+                else:
+                    dest_bucket_key = parsed_url.path
+        else:
+            parsed_url = urlparse(dest_bucket_key)
+            if parsed_url.scheme != '' or parsed_url.netloc != '':
+                raise AirflowException('If dest_bucket_name is provided, ' +
+                                       'dest_bucket_key should be relative path ' +
+                                       'from root level, rather than a full s3:// url')
+
+        if source_bucket_name is None:
+            parsed_url = urlparse(source_bucket_key)
+            if parsed_url.netloc == '':
+                raise AirflowException('Please provide source_bucket_name')
+            else:
+                source_bucket_name = parsed_url.netloc
+                if parsed_url.path[0] == '/':
+                    source_bucket_key = parsed_url.path[1:]
+                else:
+                    source_bucket_key = parsed_url.path
+        else:
+            parsed_url = urlparse(source_bucket_key)
+            if parsed_url.scheme != '' or parsed_url.netloc != '':
+                raise AirflowException('If source_bucket_name is provided, ' +
+                                       'source_bucket_key should be relative path ' +
+                                       'from root level, rather than a full s3:// url')
+
+        self.dest_bucket_key = dest_bucket_key
+        self.source_bucket_key = source_bucket_key
+        self.dest_bucket_name = dest_bucket_name
+        self.source_bucket_name = source_bucket_name
+        self.source_version_id = source_version_id
+        self.s3_conn_id = s3_conn_id
+        self.verify = verify
+
+    def execute(self, context):
+        s3_hook = S3Hook(aws_conn_id=self.s3_conn_id, verify=self.verify)
+        CopySource = {'Bucket': self.source_bucket_name,
+                      'Key': self.source_bucket_key,
+                      'VersionId': self.source_version_id}
+        response = s3_hook.get_conn().copy_object(Bucket=self.dest_bucket_name,
+                                                  Key=self.dest_bucket_key,
+                                                  CopySource=CopySource)
```

Comment:
I agree we can have `copy_object()` and `delete_object()` as methods of S3Hook(), while it would still be necessary to have two operators as wrppers of these two methods. Do you agree?

(i.e., we put the logics  into S3Hook() as methods, and then invoke them in the operators)

Reply:
Yes, logic in S3 hook, two operators.